### DISCUSSION TITLE
XTouch mini button configs

### DISCUSSION THREAD

**Discussion by mperezInnotas** [rsjaffe/MIDI2LR#717] [Q&A] [answered]
I previously had Pfixer installed with their preconfigured XTouch Mini controller via the Behringer Editor I presume.  After I reset the hardware controller to factory, I started mapping the previously mapped PFixer controls into Midi2LR commands. There is one set of buttons where I'm not sure how PFixer did it. When you clicked a button it would activate for example the Hue Color controls, then the knobs would have a color slider associated with them. Effectively the knobs had multiple controls assigned to them - the A/B assignments, then when one of these Hue, Sat, Lum buttons was depressed, they had yet another 'layer' assigned to them where the color on each knob was the individual color for Hue, Sat, Lum. 

My question is, is there something I need to setup in the Behringer Editor to tell these buttons to say on when pressed thus making another 'layer' of CC codes for the knobs?

**Comment by lovelacecreativeco** [comment, marked as answer]
The best way to go about doing this is by creating separate profiles. The easiest way to do this is to get your ENTIRE board to be the way you want it to be as standard. I had 4 profiles set up on my xtouch mini. One button for 'Exposure', one button for hue, one button for saturation, and one button for luminance. Once you've set up all buttons to your liking, and mapped 4 buttons to profiles 1-4, go to your main profile xml file and duplicate it so there are four versions. Name each one ie "00_XTOUCH_EXPOSURE.xml", "01_XTOUCH_HUE.xml", etc.

Go to file > plug-in extras > general options. Here, under the profile tab, put the corresponding profile 1-4 for each button. Profile 1 I'd recommend putting in the "00_XTOUCH_EXPOSURE.xml" file name, then go down in whichever order you did the mapping. Click OK to save your changes. 

Now press the button on your board you mapped to the HUE profile. Edit ALL of the knobs within the Midi2LR server. Do not hit ANY of your other profile buttons during this time. If you do, you'll lose all of your changes (ask me how I know lol). Once you've mapped all of the changes to each knob for each hue channel, make sure to SAVE the file, overwriting the 01_XTOUCH_HUE.xml file. Once it's saved, hit the button for the saturation profile and do the above steps again but for saturation, then save to the SATURATION.xml file.

Other tips. You always want your main profile you'll be using, like your Exposure profile, to have the first name in the file structure when ordering by name. MIDI2LR defaults to the first profile in the profiles folder in alphabetical or numerical order. That's why I always start out my main profile with 00_{your profile name here}.xml as the syntax, then any additional profiles I put as a numerically higher number at the start of the file name.

Hope this helps.

**Reply by mperezInnotas** [reply]
This is great information!  As I rewatched a few videos I started to realize the profiles was how you assign multiple functions to the same button.   One question I have is for example, assuming I hit the button i assigned for HUE profile and make a few changes there, but then I want to go back to basic stuff I've done like exposure, contrast, etc etc. do I assign a button for the basic or main adjustments so that it switches to that one (like your 00_XTOUCH_EXPOSURE.xml?). Also if for example I have the next and previous photos assigned to buttons on the 00_XTOUCH_EXPOSURE.xml file do I need to duplicate that to the other profile xml files? For example, I switched to hue made an adjustment and just one to hit the next button. Or do you have to hit the profile that the next/prev buttons are assigned to?  Basically what I'm wondering is if the HUE/SAT/LUM profiles only deal with the knobs and I want the buttons to do the same thing in these profiles as say the EXPOSURE profile do I have to duplicate them in those 3 profiles too?

**Reply by lovelacecreativeco** [reply]
> do I assign a button for the basic or main adjustments so that it switches to that one (like your 00_XTOUCH_EXPOSURE.xml?)

Precisely this. And this is why I said in my last comment to create your main profile. COMPLETELY set it up the way you want it, especially if you're copying the pfixer setup. That way all of your other buttons will remain the same, even when you change profiles. The buttons you assign to everything will remain the same on each duplicated xml file, so even though your knobs will change to hue/sat/lum, you'll still be able to pick photos, go next/prev, etc... and you won't need to remap them again since you're taking a duplicated file and just remapping the knobs to be hue/sat/lum/etc...

Please let me know if this makes sense.

**Reply by mperezInnotas** [reply]
Appreciate the help.  I'll read a little more on the docs regarding the file names and order. I was under the impression that the plugin mappings under Profiles: Modules defaults to the profile xml file you specify for the module you are starting in or move back to. I can see now why you'd want to use profiles assigned to buttons to effectively remap buttons for different sections like the local adjustments, etc etc. Thanks again for the assistance.

**Comment by onkytonk** [comment]
I think you'd really benefit from my custom layout for the X-Mini! 😄 

https://github.com/rsjaffe/MIDI2LR/discussions/538

**Comment by hdstead** [comment]
I believe you.  Looking forward to getting my hands on these files.  Thanks
for again sharing these.

Cheers,

Doug

On Sat, Sep 9, 2023 at 8:11 AM onkytonk ***@***.***> wrote:

> I think you'd really benefit from my custom layout for the X-Mini! 😄
>
> #538 <https://github.com/rsjaffe/MIDI2LR/discussions/538>
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/rsjaffe/MIDI2LR/discussions/717#discussioncomment-6955928>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AL23NPHM45HBM7FW7WV3RBLXZSBLJANCNFSM6AAAAAASUQNWZQ>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***>
>


-- 
                            ''~``
                           ( o o )
+---------------------.oooO--(_)--Oooo.--------------------------+
|
| Doug Stead, Chairman:
|
|                     Tri-M Technologies Inc.,
|               Tri-M Aviation Inc. &  Galt Air Inc.
|
|                            www.tri-m.com
|
|
|                       "Courage is resistance to fear,
|                                                        mastery of fear ,
|                                                  — not absence of fear."
|                           Mark Twain
|
|                         .oooO
|                          (  )     Oooo.
+---------------------------\ (--- (   )-------------------------+
                             \_)    ) /
                                   (_/

Perpetrators, collaborators, bystanders, victims: we can be clear about
three of these categories. The bystanders, however, act as a fulcrum and
"All that is necessary for the triumph of evil is that good people do
nothing".  If you are not a part of the solution, you are part of the
problem!  With apologies to Edmund Burke (1729 ~ 1797).

Confidentiality Notice: This e-mail and any files transmitted with it are
solely intended for the use of the recipient(s) and may contain information
that is confidential and privileged. If you receive this e-mail in error,
please advise me by return e-mail immediately. Please also disregard the
contents of the e-mail, deleting it and destroying all copies immediately.
